build/patches/add-user-scripts.v5.patch

```diff
@@ -0,0 +1,9810 @@
++      types, false /*use_media_capture*/);
++
++  dialog_->SelectFile(
++      ui::SelectFileDialog::SELECT_OPEN_FILE,
++      base::string16() /* dialog title*/, suggested_name, &allowed_file_info,
++      0 /* file type index */, std::string() /* default file extension */,
++      nativeWindow,
++      &accept_types /* params */);
++}
++
++
++void LoadScriptFromPathOnFileTaskRunner(
++  const base::FilePath& path,
++  const std::string& display_name,
++  UserScriptLoader::LoadSingleScriptCallback callback ) {
++  DCHECK(GetUserScriptsFileTaskRunner()->RunsTasksInCurrentSequence());
++
++  std::unique_ptr<UserScript> userscript(new UserScript());
++  base::string16 error;
++  bool result = LoadUserScriptFromFile(path, GURL(), userscript, &error);
++
++  if(result) {
++    if (display_name.empty() == false) {
++      userscript->set_key(display_name);
++    }
++
++    LOG(INFO) << "User Script Loaded " << userscript->name() <<
++                                   "-" << userscript->version() <<
++                                   "-" << userscript->description();
++    base::FilePath destination;
++    result = GetOrCreatePath(destination);
++    if( result == false ) {
++      error = base::ASCIIToUTF16("Cannot create destination.");
++    } else {
++      destination = destination.Append(userscript->key());
++      result = base::CopyFile(path, destination);
++      if (result == false) {
++        error = base::ASCIIToUTF16("Copy error.");
++      }
++    }
++  } else {
++    LOG(INFO) << "User Script Load error " << error;
++  }
++
++  const std::string string_error = base::UTF16ToASCII(error);
++
++  content::GetUIThreadTaskRunner({base::TaskPriority::USER_BLOCKING})
++      ->PostTask(FROM_HERE,
++                 base::BindOnce(std::move(callback), result,
++                                std::move(string_error)));
++}
++
++void UserScriptLoader::TryToInstall(const base::FilePath& script_path) {
++  base::string16 file_display_name;
++  base::MaybeGetFileDisplayName(script_path, &file_display_name);
++
++  std::string display_name = script_path.BaseName().value();
++  if (base::IsStringASCII(file_display_name))
++    display_name = base::UTF16ToASCII(file_display_name);
++
++  GetUserScriptsFileTaskRunner()->PostTask(
++      FROM_HERE,
++      base::BindOnce(
++        &LoadScriptFromPathOnFileTaskRunner,
++        script_path, display_name,
++        base::BindOnce(
++            &UserScriptLoader::LoadScriptFromPathOnFileTaskRunnerCallback,
++            weak_factory_.GetWeakPtr()
++        )
++      ));
++}
++
++void UserScriptLoader::FileSelected(
++    const base::FilePath& path, int index, void* params) {
++  LOG(INFO) << "UserScriptLoader::FileSelected " << path;
++
++  UserScriptLoader::TryToInstall(path);
++}
++
++void UserScriptLoader::LoadScriptFromPathOnFileTaskRunnerCallback(
++              bool result, const std::string& error) {
++  // LOG(INFO) << "Return from LoadScriptFromPathOnFileTaskRunnerCallback "
```

Comment:
Please uncomment or delete this line.

Reply:
fixed in https://github.com/bromite/bromite/commit/2812a0f9a76a7b3052b7a408344095680e7244ff